core/src/main/scala/caliban/schema/ObjectFieldResolver.scala

```diff
@@ -0,0 +1,36 @@
+package caliban.schema
+
+import caliban.execution.Field
+import caliban.schema.Step.{ MetadataFunctionStep, ObjectStep }
+
+import scala.collection.immutable.HashMap
+
+final private class ObjectFieldResolver[R, A](
+  objectName: String,
+  fields: Iterable[(String, A => Step[R])]
+) {
+
+  private val fieldsMap: java.util.HashMap[String, A => Step[R]] = {
+    val map = new java.util.HashMap[String, A => Step[R]]()
+    fields.foreach { case (name, resolve) => map.put(name, resolve) }
+    map
```

Comment:
Could we use a standard Scala map here? Doesn't it have a constructor that takes an iterable?

Reply:
So the main driver for going with a Java HashMap here was to avoid the wrapping / unwrapping into `Option` when calling `.get`. I found that it offers a (small) performance increase and reduction in gc allocations.

Although I do think it's cleaner to just use `immutable.HashMap.from(fields)`, wdut?